rclpy/rclpy/clock.py

```diff
@@ -57,6 +143,45 @@ def now(self):
             nanoseconds=_rclpy.rclpy_time_point_get_nanoseconds(time_handle),
             clock_type=self.clock_type)
 
+    def create_jump_callback(self, threshold, *, pre_callback=None, post_callback=None):
+        """
+        Create callback handler for clock time jumps.
+
+        The callbacks must remain valid as long as the returned JumpHandler is valid.
+        A callback should execute as quick as possible and must not block when called.
+        If a callback raises then no time jump callbacks added after it will be called.
+
+        :param threshold: Criteria for activating time jump.
+        :param pre_callback: Callback to be called before new time is set.
+        :param post_callback: Callback to be called after new time is set accepting
+        :rtype: :class:`rclpy.clock.JumpHandler`
+        """
+        original_callback = post_callback
+
+        def callback_shim(jump_dict):
+            nonlocal original_callback
+            clock_change = None
+            if 'RCL_ROS_TIME_NO_CHANGE' == jump_dict['clock_change']:
+                clock_change = ClockChange.ROS_TIME_NO_CHANGE
+            elif 'RCL_ROS_TIME_ACTIVATED' == jump_dict['clock_change']:
+                clock_change = ClockChange.ROS_TIME_ACTIVATED
+            elif 'RCL_ROS_TIME_DEACTIVATED' == jump_dict['clock_change']:
+                clock_change = ClockChange.ROS_TIME_DEACTIVATED
+            elif 'RCL_SYSTEM_TIME_NO_CHANGE' == jump_dict['clock_change']:
+                clock_change = ClockChange.SYSTEM_TIME_NO_CHANGE
+            else:
+                raise ValueError('Unknown clock jump type ' + repr(clock_change))
+            duration = Duration(nanoseconds=jump_dict['delta'])
+            original_callback(TimeJump(clock_change, duration))
+
+        if post_callback is not None and callable(post_callback):
+            post_callback = callback_shim
```

Comment:
Should the `callback_shim` function only be defined when it is necessary within the `if` block?